### PR TITLE
YOYO Technical Test

### PR DESCRIPTION
**Pull request by ellechir**
Hi Zakame,

This is my techical test. Thank you!

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#YOYO Technical Test
+
+This is YOYO's technical test called API Org App. This is a simple web application that provides a simple organizer for the user to store notes and TODOs.
+
+* Instructions
+
```
Comment (zakame) [inline readme.md]: step 0 ought to be `composer install` first

File: config/database.php [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    | Default Database Connection Name
+    |--------------------------------------------------------------------------
+    |
+    | Here you may specify which of the database connections below you wish
+    | to use as your default connection for all database work. Of course
+    | you may use many connections at once using the Database library.
+    |
+    */
+
+    'default' => env('DB_CONNECTION', 'mysql'),
```
Comment (zakame) [inline config/database.php]: Could actually default to `sqlite` instead of `mysql` here; I take this is the default...

File: readme.md [hunk 3]
```diff
@@ -0,0 +1,19 @@
+#YOYO Technical Test
+
+This is YOYO's technical test called API Org App. This is a simple web application that provides a simple organizer for the user to store notes and TODOs.
+
+* Instructions
+  
+  1. Create database called 'yoyo'.
```
Comment (zakame) [inline readme.md]: Seems to conflict with what's in `config/database.php` (it currently uses the `forge` database.)

Comment (ellechir) [inline readme.md]: Shocks! stupid me wasn't able to test it using the production mode. Thank you

File: config/database.php (hunk 2 above)
Comment (ellechir) [inline config/database.php]: yes you are correct, I am used to use mysql, that's the reason why.

File: readme.md [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+  2. To create the tables needed for this app:
+    - Using terminal, go to the root directory of this app.
+    - Run 'php artisan migrate'.
+  3. To run the application, run this command 'php artisan serve --port=3000'
```
Comment (zakame) [inline readme.md]: Prior to this, one must also rebuild `.env` (e.g. from the included `.env.example`) as well as regenerate the app key for ciphers (via `php artisan key:generate`)

Comment (zakame) [inline readme.md]: Also forgot about setting up the JS/CSS side (e.g. `npm install` and `bower install`)

File: public/css/bootstrap.min.css [hunk 5]
```diff
@@ -0,0 +1 @@
+/home/chelle/rjventure/yoyo/api-org-app/bower_components/bootstrap/dist/css/bootstrap.min.css
```
Comment (zakame) [inline public/css/bootstrap.min.css]: Shouldn't be symlinked to absolute dir!

File: public/fonts [hunk 6]
```diff
@@ -0,0 +1 @@
+/home/chelle/rjventure/yoyo/api-org-app/bower_components/bootstrap/dist/fonts
```
Comment (zakame) [inline public/fonts]: ditto

File: public/js/angular.min.js [hunk 7]
```diff
@@ -0,0 +1 @@
+/home/chelle/rjventure/yoyo/api-org-app/bower_components/angular/angular.min.js
```
Comment (zakame) [inline public/js/angular.min.js]: ditto

File: public/js/bootstrap.min.js [hunk 8]
```diff
@@ -0,0 +1 @@
+/home/chelle/rjventure/yoyo/api-org-app/bower_components/bootstrap/dist/js/bootstrap.min.js
```
Comment (zakame) [inline public/js/bootstrap.min.js]: ditto

**Comment by zakame** [comment]
@ellechir there seems to be no way of converting notes to TODOs and vice versa?  It is a basic requirement in the spec:

    The app should make it easy to switch a note to a TODO entry, and vice versa.

Also, http://localhost:3000/api should be accessible (e.g. not result in 404) and can either point to an API doc or redirect to index.

**Comment by ellechir** [comment]
My bad.
- The app should make it easy to switch a note to a TODO entry, and vice versa.

My understanding on this was the same with the #6 item.

File: resources/views/todos.blade.php [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+                    <h4>@{{ todo }}</h4>
+                    <div class="row">
+                        <div class="col-md-7">
+                            <div class="entry-list">
+                                <h4>Entries
+                                    <span ng-show="entries.length">( @{{getRemaining(entries)}} )</span>
+                                </h4>
+                                <div class="checkbox"
+                                    ng-repeat="entry in entries">
+                                    <label ng-click="updateCompletedEntry(entry.id)">
```
Comment (tedmdelacruz) [inline resources/views/todos.blade.php]: TODO entries are not marked as done, `updateCompletedEntry()`  function doesn't seem to exist

**Pull request closed without merging**